### ISSUE TITLE
Swift + DoxyDoxygen

### ISSUE THREAD

**Issue by clozach** [20Tauri/DoxyDoxygen#7]
DoxyDoxygen works fine for me with, for example, JavaScript. But when I position the cursor at the top of a function in a .swift file and press alt+command-q, I get:

"[doxy] cannot detect function… command aborted"

Any idea how to fix this?

**Comment by 20Tauri** [comment]
Swift is not natively included in sublime text.
I have tested it with the the package called "swift" (but i have notice a poor grammar)
You probably use another one.
- To help me to fix it, please go on a function name
- Open the console
- Enter "view.scope_name(view.sel()[0].begin())"
- Add console output in this issue

Thanks you

**Comment by clozach** [comment]
Ah, my mistake. It doesn't sound like you meant to imply support for Apple's Swift language. Perhaps a note in the ReadMe would be useful to prevent others from experiencing the same confusion. (Unless of course it's pretty easy to add Swift language support, which would be awesome!)

For what it's worth, here's the output from running the `scope_name` command per your instructions:

'source.swift meta.import.swift keyword.other.import.swift '

Thanks for the quick response!

**Comment by 20Tauri** [comment]
README have been added.
You probably don't use it, but "Rust" have the same problem

I will probably also improve integrated debug agent to provide "scope" informations

Thanks you for using the evolution branch

**Comment by clozach** [comment]
Ah, I'd actually misunderstood you the first time. You _do_ support Apple's Swift. It took me awhile to confirm this because:
1. I'd installed "DoxyDoxygen" rather than "DoxyDoxygen (evolution)"
2. While I already did have the "Swift" package installed, I hadn't restarted Sublime Text
3. Even after restarting, I encountered a bug whereby the `static` keyword breaks DoxyDoxygen. Will write that up separately. :)

Cheers!
